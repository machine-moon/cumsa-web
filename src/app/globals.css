@import "tailwindcss";

/* ========== CSS Variables ========== */
:root {
  --background: #edfaff;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --green: #169845ff;
  --header-top: #123a66;
}

/* ========== Base Styles ========== */
body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

/* ========== Links ========== */
.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
  transition: color 150ms ease;
}

.link-cta:hover {
  color: var(--red);
}

/* ========== Buttons ========== */
.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

/* ========== Accessibility ========== */
:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

/* ========== Layout Containers ========== */
.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

/* ========== Transition Utility ========== */
.transition-base {
  transition: all 200ms ease-out;
}

/* ========== Cards ========== */
.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

/* ========== Typography ========== */
.title {
  color: var(--red);
}

.typewriter-cursor {
  color: var(--blue);
}

/* ========== Icon and Logo ========== */
.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank.png");
  mask-image: url("/CU-MSA-LOGO-blank.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}

/* ========== Special Surfaces ========== */
.portal-glow {
  width: 100%;
  max-width: 72rem;
  border-radius: 0.75rem;
  box-shadow:
    0 4px 32px 0 #5aced880,
    0 0 0 2px #5aced850;
  background: #fff;
  border: none;
  padding: 0;
}

.frost-card {
  background: color-mix(in oklab, var(--header-top) 45%, transparent);
  border: 1px solid rgba(255, 255, 255, 0.18);
  border-radius: 1rem;
  box-shadow:
    0 20px 70px rgba(10, 22, 40, 0.25),
    0 2px 6px rgba(0, 0, 0, 0.05);
  backdrop-filter: blur(10px) saturate(120%);
}

/* ========== Hero Section ========== */
.hero-overlay {
  background:
    radial-gradient(1000px 500px at 20% 20%, rgba(90, 206, 216, 0.18), transparent 60%),
    radial-gradient(800px 420px at 80% 30%, rgba(199, 50, 33, 0.1), transparent 60%),
    linear-gradient(to bottom, rgba(18, 58, 102, 0.25), rgba(18, 58, 102, 0.5));
}

.hero-glow {
  position: absolute;
  pointer-events: none;
  filter: blur(48px);
  opacity: 0.5;
}

.hero-glow-1 {
  width: 40vw;
  height: 40vh;
  left: -10vw;
  top: 20vh;
  background: radial-gradient(circle at center, rgba(90, 206, 216, 0.28), transparent 60%);
}

.hero-glow-2 {
  width: 35vw;
  height: 35vh;
  right: -8vw;
  bottom: 10vh;
  background: radial-gradient(circle at center, rgba(199, 50, 33, 0.14), transparent 60%);
}

@media (max-width: 640px) {
  .hero-glow {
    display: none;
  }
}

/* ========== Animations ========== */
/* Fade / Slide / Spin / Float / Sparkle */
@keyframes fade-in {
  from {
    opacity: 0;
    transform: translateY(20px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
}

@keyframes slide-in-left {
  from {
    opacity: 0;
    transform: translateX(-30px);
  }

  to {
    opacity: 1;
    transform: translateX(0);
  }
}

@keyframes slide-in-right {
  from {
    opacity: 0;
    transform: translateX(30px);
  }

  to {
    opacity: 1;
    transform: translateX(0);
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@keyframes float {
  0%,
  100% {
    transform: translateY(0);
  }

  50% {
    transform: translateY(-10px);
  }
}

@keyframes sparkle {
  0%,
  100% {
    opacity: 0;
    transform: scale(0.5);
  }

  50% {
    opacity: 1;
    transform: scale(1);
  }
}

.animate-fade-in {
  animation: fade-in 0.6s ease-out;
}

.animate-slide-in-left {
  animation: slide-in-left 0.6s ease-out 0.2s both;
}

.animate-slide-in-right {
  animation: slide-in-right 0.6s ease-out 0.4s both;
}

.animate-spin {
  animation: spin 1s linear infinite;
}

.animate-float {
  animation: float 3s ease-in-out infinite;
}

.animate-sparkle {
  animation: sparkle 2s ease-in-out infinite;
}

/* ========== Custom Scrollbar ========== */
.custom-scroll {
  scrollbar-width: thin;
  scrollbar-color: var(--blue) transparent;
}

.custom-scroll::-webkit-scrollbar {
  width: 8px;
}

.custom-scroll::-webkit-scrollbar-track {
  background: transparent;
}

.custom-scroll::-webkit-scrollbar-thumb {
  background: color-mix(in oklab, var(--blue) 75%, white);
  border-radius: 8px;
  border: 2px solid transparent;
  background-clip: content-box;
}

.custom-scroll::-webkit-scrollbar-thumb:hover {
  background: var(--blue);
}
