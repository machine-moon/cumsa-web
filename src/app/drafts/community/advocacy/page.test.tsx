import { test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import AdvocacyPage from "./page";

test("advocacy page snapshot", () => {
  const { asFragment } = render(<AdvocacyPage />);
  expect(asFragment()).toMatchSnapshot();
});
