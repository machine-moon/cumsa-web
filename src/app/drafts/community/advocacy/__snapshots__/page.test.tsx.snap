// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`advocacy page snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gradient-to-b from-slate-50 to-white min-h-screen"
  >
    <section
      class="relative overflow-hidden"
    >
      <div
        class="absolute inset-0 bg-gradient-to-br from-[var(--navy)] via-[var(--navy)] to-[var(--blue)]/80"
      />
      <div
        class="hero-glow hero-glow-1"
      />
      <div
        class="hero-glow hero-glow-2"
      />
      <div
        class="relative container-base py-16 md:py-20"
      >
        <div
          class="max-w-4xl mx-auto text-center"
          style="opacity: 0; transform: translateY(20px);"
        >
          <h1
            class="text-4xl md:text-5xl font-bold text-white mb-6"
          >
            Standing in Solidarity
          </h1>
          <p
            class="text-lg md:text-xl text-white/90 leading-relaxed mb-6"
          >
            As the Muslim Student Association, we stand in unwavering solidarity with our brothers and sisters who face oppression and injustice around the world.
          </p>
          <div
            class="bg-white/10 backdrop-blur-sm rounded-xl p-6 border border-white/20"
          >
            <p
              class="text-white/95 text-base md:text-lg italic leading-relaxed"
            >
              “And what is [the matter] with you that you fight not in the cause of Allāh and [for] the oppressed among men, women and children who say, “Our Lord, take us out of this city of oppressive people and appoint for us from Yourself a protector and appoint for us from Yourself a helper”
            </p>
            <p
              class="text-white/80 mt-3 text-sm"
            >
              — Quran 4:75 (Saheeh International)
            </p>
          </div>
        </div>
      </div>
    </section>
    <div
      class="container-base py-12"
    >
      <div
        class="max-w-6xl mx-auto"
        style="opacity: 0;"
      >
        <div
          class="text-center mb-12"
          style="opacity: 0; transform: translateY(30px);"
        >
          <h2
            class="text-3xl font-bold text-[var(--navy)] mb-4"
          >
            Current Crises
          </h2>
          <p
            class="text-gray-600 max-w-2xl mx-auto"
          >
            Learn about ongoing situations affecting Muslim communities worldwide and discover how you can make a difference.
          </p>
        </div>
        <div
          class="grid gap-6 md:grid-cols-2 lg:grid-cols-2 mb-16"
        >
          <div
            style="opacity: 0; transform: translateY(30px);"
          >
            <button
              class="w-full text-left bg-white rounded-2xl shadow-lg hover:shadow-2xl transition-all duration-300 p-6 border border-gray-100 hover:border-[var(--blue)] hover:scale-[1.02] group"
            >
              <div
                class="flex items-start gap-4"
              >
                <div
                  class="text-5xl flex-shrink-0"
                >
                  🇵🇸
                </div>
                <div
                  class="flex-1"
                >
                  <h3
                    class="text-2xl font-bold text-[var(--navy)] mb-2 group-hover:text-[var(--blue)] transition-colors"
                  >
                    Palestine
                  </h3>
                  <p
                    class="text-gray-600 mb-4"
                  >
                    Palestinians face ongoing occupation, displacement, and humanitarian crisis in Gaza and the West Bank.
                  </p>
                  <div
                    class="flex items-center text-[var(--blue)] font-medium group-hover:gap-2 transition-all"
                  >
                    Learn more 
                    <svg
                      class="ml-2 text-sm"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </button>
          </div>
          <div
            style="opacity: 0; transform: translateY(30px);"
          >
            <button
              class="w-full text-left bg-white rounded-2xl shadow-lg hover:shadow-2xl transition-all duration-300 p-6 border border-gray-100 hover:border-[var(--blue)] hover:scale-[1.02] group"
            >
              <div
                class="flex items-start gap-4"
              >
                <div
                  class="text-5xl flex-shrink-0"
                >
                  🇸🇩
                </div>
                <div
                  class="flex-1"
                >
                  <h3
                    class="text-2xl font-bold text-[var(--navy)] mb-2 group-hover:text-[var(--blue)] transition-colors"
                  >
                    Sudan
                  </h3>
                  <p
                    class="text-gray-600 mb-4"
                  >
                    Sudan faces a devastating civil war causing mass displacement, famine, and humanitarian catastrophe.
                  </p>
                  <div
                    class="flex items-center text-[var(--blue)] font-medium group-hover:gap-2 transition-all"
                  >
                    Learn more 
                    <svg
                      class="ml-2 text-sm"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </button>
          </div>
          <div
            style="opacity: 0; transform: translateY(30px);"
          >
            <button
              class="w-full text-left bg-white rounded-2xl shadow-lg hover:shadow-2xl transition-all duration-300 p-6 border border-gray-100 hover:border-[var(--blue)] hover:scale-[1.02] group"
            >
              <div
                class="flex items-start gap-4"
              >
                <div
                  class="text-5xl flex-shrink-0"
                >
                  🇨🇳
                </div>
                <div
                  class="flex-1"
                >
                  <h3
                    class="text-2xl font-bold text-[var(--navy)] mb-2 group-hover:text-[var(--blue)] transition-colors"
                  >
                    China
                  </h3>
                  <p
                    class="text-gray-600 mb-4"
                  >
                    Uyghur Muslims face systematic persecution, mass detention, and cultural genocide in Xinjiang.
                  </p>
                  <div
                    class="flex items-center text-[var(--blue)] font-medium group-hover:gap-2 transition-all"
                  >
                    Learn more 
                    <svg
                      class="ml-2 text-sm"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </button>
          </div>
          <div
            style="opacity: 0; transform: translateY(30px);"
          >
            <button
              class="w-full text-left bg-white rounded-2xl shadow-lg hover:shadow-2xl transition-all duration-300 p-6 border border-gray-100 hover:border-[var(--blue)] hover:scale-[1.02] group"
            >
              <div
                class="flex items-start gap-4"
              >
                <div
                  class="text-5xl flex-shrink-0"
                >
                  🇪🇬
                </div>
                <div
                  class="flex-1"
                >
                  <h3
                    class="text-2xl font-bold text-[var(--navy)] mb-2 group-hover:text-[var(--blue)] transition-colors"
                  >
                    Egypt
                  </h3>
                  <p
                    class="text-gray-600 mb-4"
                  >
                    Egyptians face authoritarian rule, mass political imprisonment, and severe restrictions on freedoms.
                  </p>
                  <div
                    class="flex items-center text-[var(--blue)] font-medium group-hover:gap-2 transition-all"
                  >
                    Learn more 
                    <svg
                      class="ml-2 text-sm"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
